src/opencl/opencl_utils.hpp

```diff
@@ -46,21 +46,26 @@ void getAllPlatforms(cl_platform_id*& platforms, cl_uint& num_platforms, cl_int&
     err = clGetPlatformIDs(num_platforms, platforms, NULL); if (err != CL_SUCCESS) return;
 }
 
-std::string getPlatformName(cl_platform_id& platform_id, cl_int& err) {
-
-    std::string platform_name = "";
-
-    size_t platform_name_length;
-    err = clGetPlatformInfo(platform_id, CL_PLATFORM_NAME, 0, NULL, &platform_name_length);
-    if (err != CL_SUCCESS) return platform_name;
-
-    platform_name.resize(platform_name_length);
-
-    err = clGetPlatformInfo(platform_id, CL_PLATFORM_NAME, platform_name_length,
-                            (void*)platform_name.data(), NULL);
-    if (err != CL_SUCCESS) return platform_name;;
-
-    return platform_name;
+bool isSupportedOpenClVersion(cl_platform_id& platform_id, cl_int& err) {
+    size_t platform_version_length;
+    err = clGetPlatformInfo(platform_id, CL_PLATFORM_VERSION, 0, NULL, &platform_version_length);
+    if (err != CL_SUCCESS) return false;
+
+    std::string platform_version = "";
+    platform_version.resize(platform_version_length);
+
+    err = clGetPlatformInfo(platform_id, CL_PLATFORM_VERSION, platform_version_length,
+                            (void*)platform_version.data(), NULL);
+    if (err != CL_SUCCESS) return false;
+
+    // Extracts major OpenCL version number out of string like "OpenCL 1.1 Mesa 22.0.1"
+    std::string opencl_major_version_string = platform_version.substr(

```

Comment:
This doesn't seem to be a stable solution. I would recommend using regex instead. It's clearer and can't throw an exception. At the very least, the comment should contain a reference to CL_PLATFORM_VERSION  with `OpenCL<space><major_version.minor_version><space><platform-specific information>` string.

Reply:
Format is defined in OpenCL spec, hence we can rely on it: https://www.khronos.org/registry/OpenCL/sdk/2.0/docs/man/xhtml/clGetPlatformInfo.html